### ISSUE TITLE
@vanilla-extract does not support features from typescript 4.7+

### ISSUE THREAD

**Issue by mc-petry** [vanilla-extract-css/vanilla-extract#784]
### Describe the bug

For example: vanilla-extract does not support typescript [optional variance](https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/#optional-variance-annotations-for-type-parameters). (in out in types)

You will get some thing like this:
```
[plugin:vanilla-extract] Build failed with 1 error:
/button/index.tsx:10:21: error: Expected identifier but found "in"
```

Seems the same reason for #739 . Do we need to update the typescript?

### Reproduction

https://github.com/mc-petry/vanilla-extract-typescript

### System Info

```shell
System:
    OS: macOS 12.4
    CPU: (10) arm64 Apple M1 Pro
    Memory: 76.14 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.14.2 - /usr/local/bin/node
    Yarn: 1.22.15 - /usr/local/bin/yarn
    npm: 8.5.0 - /usr/local/bin/npm
  Browsers:
    Chrome: 104.0.5112.79
    Firefox: 103.0.1
    Safari: 15.5
  npmPackages:
    @vanilla-extract/css: ^1.7.2 => 1.7.2 
    @vanilla-extract/recipes: 0.2.5 => 0.2.5 
    @vanilla-extract/sprinkles: ^1.4.1 => 1.4.1 
    @vanilla-extract/vite-plugin: ^3.3.0 => 3.3.0 
    vite: ^3.0.7 => 3.0.7
```

### Used Package Manager

pnpm

### Validations

- [X] Check that there isn't [already an issue](https://github.com/seek-oss/vanilla-extract/issues) that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by benjervis** [comment]
The fix here is actually an esbuild version bump, looks like they implemented it in [0.14.31](https://github.com/evanw/esbuild/releases/tag/v0.14.31), so I don't think it's related to #739. We'll have a look into this though, and just make sure that none of those breaking changes between 11 and 14 are going to be a problem.

**Comment by askoufis** [comment]
Fixed in #973
